redbot/cogs/filter/filter.py

```diff
@@ -279,6 +283,10 @@ def __unload(self):
         else:
             await ctx.send(_("Names and nicknames will now be filtered."))
 
+    def invalidate_cache(self, guild: discord.Guild, channel: discord.TextChannel = None):
+        """ Invalidate a cached pattern"""
+        self.pattern_cache.pop((guild, None), None)
```

Comment:
Needs support for invalidating the cache in a channel as well

Reply:
Toby, the key is a tuple there. That's intended.

Reply:
Why bother having a channel parameter if it's unused?

Reply:
Fixed....